flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerStaticFileServerHandler.java

```diff
@@ -266,6 +324,18 @@ private void respondWithFile(ChannelHandlerContext ctx, HttpRequest request, Str
         }
     }
 
+    static String extractJobId(String requestPath) {
+        if (StringUtils.isNullOrWhitespaceOnly(requestPath)
+                || !requestPath.matches("^/jobs/.{32}\\.json$")) {
+            return null;
+        }
+        String secondPath = requestPath.split("/")[2];
+        if (StringUtils.isNullOrWhitespaceOnly(secondPath) || secondPath.length() < 32) {
+            return null;
+        }
+        return secondPath.substring(0, 32);
```

Comment:
Too many 32, the 32 should be defined as constant.

Reply:
Done